### ISSUE TITLE
attributionEnabled crashes the application on iOS

### ISSUE THREAD

**Issue by tyao1** [nitaliano/react-native-mapbox-gl#291]
if set the attributionButtonIsHidden to true, the application would crash.
But it works well on Android.

**Comment by urbancvek** [comment]
All you have to do is add a line to Info.plist in Xcode. 
1. Open Info.plist inside Xcode
2. Add new line MGLMapboxMetricsEnabledSettingShownInApp
3. Type should be boolean
4. Value should be YES

<img width="612" alt="screen shot 2016-02-27 at 01 46 30" src="https://cloud.githubusercontent.com/assets/12081272/13369350/f4a36734-dcf3-11e5-87d0-9b2b60197b85.png">

**Comment by friedbunny** [comment]
Hrm, does RNMBGL not print the error message here? **This crash is intentional** and should show this very informative message:

> **Telemetry opt-out missing**
>  End users must be able to opt out of Mapbox Telemetry in your app, either inside Settings (via Settings.bundle) or inside this app.
> 
>  By default, this opt-out control is included as a menu item in the attribution action sheet.
> 
>  If you reimplement the opt-out control inside this app, disable this assertion by setting MGLMapboxMetricsEnabledSettingShownInApp to YES in Info.plist.
> 
>  See https://www.mapbox.com/ios-sdk/#telemetry_opt_out for more information.
> 
>  Additionally, by hiding this attribution control you agree to display the required attribution elsewhere in this app.

In other words, **you are required to comply with the [Mapbox Terms of Service](https://www.mapbox.com/tos/#[YmpaYuta])**, which state that:
1. You must credit OpenStreetMap.
2. You must credit Mapbox and, if you're on the Starter or Standard plans, show our branding.
3. You must offer your users the means to opt-out of [Mapbox Telemetry](https://mapbox.com/telemetry/).

If you hide the attribution button, you are responsible for providing proper attribution elsewhere in your app. You are also then responsible for providing your users with a way to opt-out of Mapbox Telemetry, which is most easily accomplished by adding [this Settings.bundle](https://www.mapbox.com/help/data/ios/Settings.bundle.zip) to your app.

Only after complying with these terms of service may you then use the  `MGLMapboxMetricsEnabledSettingShownInApp` key to disable this crash/error. If you add the suggested Settings.bundle to your app, you won't even need to set this key.

**Comment by tyao1** [comment]
Oops, sorry I didn't notice that, thank you for the information!

**Comment by lucaswoj** [comment]
> Hrm, does RNMBGL not print the error message here? This crash is intentional and should show this very informative message:

RNMBGL does not print any error messages in this situation. Re-opening to track improvement of error reporting.

**Comment by franky47** [comment]
I had the same issue, and ended up here to see the resolution.

A little note in the documentation next to the `attributionButtonIsHidden` flag would be appreciated, more than just a link to the terms & conditions.

@friedbunny the crash is silent if the app is not attached to Xcode or log-ios.

**Comment by friedbunny** [comment]
This is still tripping up users — https://github.com/mapbox/react-native-mapbox-gl/issues/860.

**Comment by nitaliano** [comment]
Updated the title for anyone wondering this was called `attributionButtonIsHidden` in the unsupported version(v5) and in the supported version(v6) this property is called `attributionEnabled`

**Comment by nitaliano** [comment]
docs have been updated with this PR https://github.com/mapbox/react-native-mapbox-gl/pull/942 and it's been merged into master closing this out
